### ISSUE TITLE
printpreview assumes the notebook is in the current working directory

### ISSUE THREAD

**Issue by achampion** [ipython-contrib/jupyter_contrib_nbextensions#1085]
Printpreview fails if you have changed the working directory of your notebook, e.g.

    %cd subdirectory

Now printpreview fails with:

    [NbConvertApp] WARNING | pattern 'notebook.ipynb' matched no files

And just prints out the help message to `nbconvert` in the log.

**Comment by juhasch** [comment]
It is not so easy to get the path right, because the current implementation of `printview` starts an `jupyter nbconvert` command in the current kernel, and possibly as you noticed in the wron directory. Fixing would be possible. I think the better solution is to wait for https://github.com/jupyter/notebook/pull/2413.

**Comment by achampion** [comment]
Agree, it would be better to have jupyter do the conversion vs the python kernel doing it. As you mentioned while it is easy to get the path of the notebook relative to the jupyter root, getting that jupyter root in the python kernel not so easy.

**Comment by michalkahle** [comment]
Printview also doesn't work with R kernel (or any other than Python kernel) for the reason mentioned above (tries to execute Python code in current kernel to convert the notebook).
